content/en/ebpf.md

```diff
@@ -0,0 +1,40 @@
+---
+title: eBPF
+status: Completed
+category: architecture
+---
+
+## What it is
+
+eBPF, or extended Berkeley Packet Filter, is a technology that allows small, sandboxed programs or scripts to run in the kernel space of a Linux system without having to change the kernel's source code or load Linux kernel modules.
+
+A Linux system has two spaces: the kernel and the user space. 
+The kernel represents the operating system's core and is the only part 
+with unlimited access to the hardware. 
+
+Applications reside in the user space, and when they need higher permissions, 
+they send a request to the kernel.
+For applications that require more flexibility, such as direct hardware 
+access, the kernel can be extended via what is known as the "Linux 
+kernel modules" approach. This approach extends the kernel's default functionality
+ allowing applications deeper access to the underlying components. 
+ Unfortunately, this could introduce security risks as well.
+
+## Problem it addresses
+Typically, applications run in the user space, and if the application requires some privileges from the kernel (e.g., to access some hardware), 
+it requests it from the kernel via a so-called "system call."  
+In most cases, this approach works just fine. However, there are instances where developers require more flexibility for low-level system access.
+Observability, security, and networking features are some of the ideal examples to be implemented within the kernel space.
+One of the possibilities to reach such a goal is to use Linux kernel modules, which allow extending the kernel base without adding directly to the kernel source code. 
+While using Linux kernel modules bring the mentioned benefits, it adds to the security risks because they operate within the kernel space and can make the kernel crash. 
+Kernel modules have elevated privileges and direct access to system resources, which, if not properly secured, can be exploited by attackers.
```

Comment:
Did I understand this correctly? There are two risks: 1) the risk of crashing the kernel, and 2) the risk of giving attackers direct access to system resources 
```suggestion
Additionally, kernel modules have elevated privileges and direct access to system resources. And if not properly secured, attackers can exploit these.
```

Reply:
Also this seems to be correct.